### ISSUE TITLE
Route generation for articles stopped working after upgrading to Sulu 2.5.12

### ISSUE THREAD

**Issue by rogamoore** [sulu/sulu#7270]
| Q | A
| --- | ---
| Sulu Version | 2.5.12
| PHP Version | 8.3.1
| DB Version | -
| Browser Version | -

Generation of routes fails for articles after upgrading to Sulu 2.5.12 (from Sulu 2.5.11) with no other dependency changes.

# Actual Behavior

The request to `/admin/api/routes?action=generate` fails with HTTP 500:

Request:
```
POST /admin/api/routes?action=generate HTTP/1.1

{
    "parts": {
        "title": "dummy title"
    },
    "resourceKey": "articles",
    "locale": "en",
    "id": null,
    "entityClass": null,
    "routeSchema": "/{translator.trans('article')}/{implode('-', object)}"
}
```

Response:
```
HTTP/1.1 500 Internal Server Error

{
    "code": 0,
    "message": "implode(): Argument #2 ($array) must be of type ?array, Sulu\\Bundle\\RouteBundle\\Model\\ResourceLocatorParts given",
    "errors": [
        "TypeError: implode(): Argument #2 ($array) must be of type ?array, Sulu\\Bundle\\RouteBundle\\Model\\ResourceLocatorParts given in /var/www/sulu/vendor/symfony/expression-language/ExpressionFunction.php:87\nStack trace:\n#0 /var/www/sulu/vendor/symfony/expression-language/ExpressionFunction.php(87): implode()\n#1 /var/www/sulu/vendor/symfony/expression-language/Node/FunctionNode.php(50): Symfony\\Component\\ExpressionLanguage\\ExpressionFunction::Symfony\\Component\\ExpressionLanguage\\{closure}()\n#2 /var/www/sulu/vendor/symfony/expression-language/ExpressionLanguage.php(59): Symfony\\Component\\ExpressionLanguage\\Node\\FunctionNode->evaluate()\n#3 /var/www/sulu/vendor/sulu/sulu/src/Sulu/Bundle/RouteBundle/Generator/SymfonyExpressionTokenProvider.php(69): Symfony\\Component\\ExpressionLanguage\\ExpressionLanguage->evaluate()\n#4 /var/www/sulu/vendor/sulu/sulu/src/Sulu/Bundle/RouteBundle/Generator/RouteGenerator.php(51): Sulu\\Bundle\\RouteBundle\\Generator\\SymfonyExpressionTokenProvider->provide()\n#5 /var/www/sulu/vendor/sulu/sulu/src/Sulu/Bundle/RouteBundle/Controller/RouteController.php(218): Sulu\\Bundle\\RouteBundle\\Generator\\RouteGenerator->generate()\n#6 /var/www/sulu/vendor/sulu/sulu/src/Sulu/Bundle/RouteBundle/Controller/RouteController.php(110): Sulu\\Bundle\\RouteBundle\\Controller\\RouteController->generateUrlResponse()\n#7 /var/www/sulu/vendor/symfony/http-kernel/HttpKernel.php(181): Sulu\\Bundle\\RouteBundle\\Controller\\RouteController->postAction()\n#8 /var/www/sulu/vendor/symfony/http-kernel/HttpKernel.php(76): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()\n#9 /var/www/sulu/vendor/symfony/http-kernel/Kernel.php(197): Symfony\\Component\\HttpKernel\\HttpKernel->handle()\n#10 /var/www/sulu/public/index.php(77): Symfony\\Component\\HttpKernel\\Kernel->handle()\n#11 {main}"
    ]
}
```

# Expected Behavior

Does not fail 🙂 

# Steps to Reproduce

1. Create a new article
2. Enter a title and click away
3. The URL generation is triggered
4. Check dev tools for the failed request

# Workaround

Manually enter the URL.

# Possible Solutions

The class `ResourceLocatorParts` was introduced here https://github.com/sulu/sulu/pull/7236.

Note that the routeSchema `{implode('-', object)}` is the default from the installation instructions (except the translation): https://github.com/sulu/SuluArticleBundle/blob/2.5/Resources/doc/installation.md#configure-suluarticlebundle-and-sulu-core

**Comment by wachterjohannes** [comment]
@rogamoore thanks for reporting. I already found the issue and add a fix for it :) https://github.com/sulu/sulu/pull/7271 feel free to test it!

**Comment by rogamoore** [comment]
@wachterjohannes That was quick! Thanks a lot! I can confirm it's fixed with #7271 🎉

**Comment by wachterjohannes** [comment]
@rogamoore next release is your friend :) thanks for reporting
